docs/guide/lifecycle.md

```diff
@@ -0,0 +1,21 @@
+---
+layout: default
+title: Lifecycle
+parent: Building ViewComponents
+---
+
+# Lifecycle
```

Comment:
I think a diagram would be awesome to have on this page. Thoughts on opening an issue for one? Maybe something similar to https://twitter.com/dan_abramov/status/981712092611989509/photo/1 would be nice.

Reply:
I'd love to see `_after_compile` here too :)

Reply:
Filed in https://github.com/github/view_component/issues/505#issuecomment-812577891